tst/EnergyPlus/unit/AirLoopHVACDOAS.unit.cc

```diff
@@ -4304,4 +4305,161 @@ TEST_F(EnergyPlusFixture, AirLoopHVACDOAS_TestOACompOutletNodeIndex)
     EXPECT_EQ(state->dataAirLoop->OutsideAirSys(1).OutletNodeNum(2), 23);
 }
 
+TEST_F(EnergyPlusFixture, AirLoopHVACDOAS_TestGetDesignDayConditions)
+{
+    AirLoopHVACDOAS::AirLoopDOAS thisDOAS;
+
+    int constexpr summerDesignDayTypeIndex(9);
+    int constexpr winterDesignDayTypeIndex(10);
+
+    // set up design days that are incrementally a larger peak such that the latter ones should be chosen
+    state->dataWeatherManager->DesDayInput.allocate(4);
+    state->dataWeatherManager->DesDayInput(1).DayType = summerDesignDayTypeIndex;
+    state->dataWeatherManager->DesDayInput(1).MaxDryBulb = 27.0;
+    state->dataWeatherManager->DesDayInput(1).HumIndType = WeatherManager::DDHumIndType::HumRatio;
+    state->dataWeatherManager->DesDayInput(1).HumIndValue = 0.008;
+    state->dataWeatherManager->DesDayInput(2).DayType = summerDesignDayTypeIndex;
+    state->dataWeatherManager->DesDayInput(2).MaxDryBulb = 31.0;
+    state->dataWeatherManager->DesDayInput(2).HumIndType = WeatherManager::DDHumIndType::HumRatio;
+    state->dataWeatherManager->DesDayInput(2).HumIndValue = 0.01;
+    state->dataWeatherManager->DesDayInput(3).DayType = winterDesignDayTypeIndex;
+    state->dataWeatherManager->DesDayInput(3).MaxDryBulb = 9.0;
+    state->dataWeatherManager->DesDayInput(3).HumIndType = WeatherManager::DDHumIndType::HumRatio;
+    state->dataWeatherManager->DesDayInput(3).HumIndValue = 0.006;
+    state->dataWeatherManager->DesDayInput(4).DayType = winterDesignDayTypeIndex;
+    state->dataWeatherManager->DesDayInput(4).MaxDryBulb = 6.0;
+    state->dataWeatherManager->DesDayInput(4).HumIndType = WeatherManager::DDHumIndType::HumRatio;
+    state->dataWeatherManager->DesDayInput(4).HumIndValue = 0.004;
+
+    // test peak conditions used for sizing
+    thisDOAS.GetDesignDayConditions(*state);
+
+    EXPECT_NEAR(thisDOAS.SizingCoolOATemp, state->dataWeatherManager->DesDayInput(2).MaxDryBulb, 0.000001);
+    EXPECT_NEAR(thisDOAS.SizingCoolOAHumRat, state->dataWeatherManager->DesDayInput(2).HumIndValue, 0.000001);
+    EXPECT_NEAR(thisDOAS.HeatOutTemp, state->dataWeatherManager->DesDayInput(4).MaxDryBulb, 0.000001);
+    EXPECT_NEAR(thisDOAS.HeatOutHumRat, state->dataWeatherManager->DesDayInput(4).HumIndValue, 0.000001);
+
+    AirLoopHVACDOAS::AirLoopDOAS anotherDOAS;
```

Comment:
This is local to the unit test, so it will be destroyed when the text exits (said the reviewer who really doesn't know for sure, but thinks this is true).